pandas/core/internals/construction.py

```diff
@@ -677,10 +657,22 @@ def sanitize_array(data, index, dtype=None, copy=False,
     return subarr
 
 
-def _try_cast(arr, take_fast_path, dtype, copy, raise_cast_failure):
-
+def _try_cast(arr, dtype, copy, raise_cast_failure):
+    """
+    Convert input to numpy ndarray and optionally cast to a given dtype.
+
+    Parameters
+    ----------
+    arr : array-like
```

Comment:
Any restriction on ndim?  Some of these functions are only called via the dataframe constructor.